internal/core/src/query/SearchOnSealed.cpp

```diff
@@ -73,6 +77,84 @@ SearchOnSealedIndex(const Schema& schema,
     search_result.unity_topK_ = topK;
 }
 
+void
+SearchOnSealed(const Schema& schema,
+               std::shared_ptr<ChunkedColumnBase> column,
+               const SearchInfo& search_info,
+               const void* query_data,
+               int64_t num_queries,
+               int64_t row_count,
+               const BitsetView& bitview,
+               SearchResult& result) {
+    auto field_id = search_info.field_id_;
+    auto& field = schema[field_id];
+
+    // TODO(SPARSE): see todo in PlanImpl.h::PlaceHolder.
+    auto dim = field.get_data_type() == DataType::VECTOR_SPARSE_FLOAT
+                   ? 0
+                   : field.get_dim();
+
+    query::dataset::SearchDataset dataset{search_info.metric_type_,
+                                          num_queries,
+                                          search_info.topk_,
+                                          search_info.round_decimal_,
+                                          dim,
+                                          query_data};
+
+    auto data_type = field.get_data_type();
+    CheckBruteForceSearchParam(field, search_info);
+    auto num_chunk = column->num_chunks();
+
+    SubSearchResult final_qr(num_queries,
+                             search_info.topk_,
+                             search_info.metric_type_,
+                             search_info.round_decimal_);
+
+    auto offset = 0;
+    for (int i = 0; i < num_chunk; ++i) {
+        auto vec_data = column->Data(i);
+        auto chunk_size = column->chunk_row_nums(i);
+        char bitset_data[(chunk_size + 7) / 8];
+        std::fill(bitset_data, bitset_data + sizeof(bitset_data), 0);
+        bitset::detail::ElementWiseBitsetPolicy<char>::op_copy(
```

Comment:
copy is expensive, can we avoid this copy?

Reply:
> copy is expensive, can we avoid this copy?

chunk size may can not be divisable by 8, so we have to copy it bit by bit. But if the offset can be divisable by 8 we can just use a subview.